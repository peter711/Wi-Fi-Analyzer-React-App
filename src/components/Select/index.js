import React from 'react';
import styled from 'styled-components';

const SelectWrapper = styled.div`
    display: flex;
    flex-direction: column;
    margin-bottom: 15px;

    label {
        margin-bottom: 10px;
        font-weight: bold;
        color: silver;
    }

    select {
        background-color: black;
        color: white;
        border-color: silver;
        padding: 0.5em;
        font-weight: bold;
    }
`;

class Select extends React.PureComponent {
    render() {
        const { options, label, onChange } = this.props;
        return (
            <SelectWrapper>
                <label>
                    {label}
                </label>
                <select onChange={e => {
                    const value = e.target.value;
                    onChange(value);
                }
                } value={this.props.value}>
                    {renderOptions(options)}
                </select>
            </SelectWrapper>
        );
    }
}

export default Select;

/////////////////////////////////////////////////


function renderOptions(options) {
    return (
        options.map((option, index) => renderOption(option, index))
    );
}

function renderOption({ value, text }, index) {
    return (
        <option key={index} value={value}>{text}</option>
    );
}