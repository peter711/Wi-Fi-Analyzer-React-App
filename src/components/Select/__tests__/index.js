import React from "react";
import { shallow } from "enzyme";

import Select from "../";

const options = [
  { value: 'foo', text: 'Foo' },
  { value: 'bar', text: 'Bar' }
];

describe("<Select/>", () => {
  it("should render without options", () => {
    const component = shallow(<Select />);
    expect(component).toMatchSnapshot();
  });

  it("should render options", () => {
    const component = shallow(<Select options={options} />);

    const selectOptions = component.find('option');
    expect(selectOptions).toHaveLength(2);

    expect(component).toMatchSnapshot();
  });

  it("should call onChange", () => {
    const value = 'bar';
    const onChange = jest.fn();
    const component = shallow(<Select onChange={onChange} options={options}/>);
    expect(component).toMatchSnapshot();

    component.find('select').simulate('change', {
      target: { value }
    });

    expect(onChange).toBeCalledWith(value);
  });

  it("should have label", () => {
    const label = 'Test label';
    const component = shallow(<Select label={label}/>);

    expect(component.find('label').text()).toEqual(label);
    expect(component).toMatchSnapshot();
  });
});
