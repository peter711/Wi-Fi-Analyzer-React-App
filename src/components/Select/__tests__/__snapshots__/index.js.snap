// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select/> should call onChange 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Select
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "text": "Foo",
          "value": "foo",
        },
        Object {
          "text": "Bar",
          "value": "bar",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <label />,
        <select
          onChange={[Function]}
        >
          <option
            value="foo"
          >
            Foo
          </option>
          <option
            value="bar"
          >
            Bar
          </option>
        </select>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="foo"
            >
              Foo
            </option>,
            <option
              value="bar"
            >
              Bar
            </option>,
          ],
          "onChange": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "Foo",
              "value": "foo",
            },
            "ref": null,
            "rendered": "Foo",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "Bar",
              "value": "bar",
            },
            "ref": null,
            "rendered": "Bar",
            "type": "option",
          },
        ],
        "type": "select",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <label />,
          <select
            onChange={[Function]}
          >
            <option
              value="foo"
            >
              Foo
            </option>
            <option
              value="bar"
            >
              Bar
            </option>
          </select>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value="foo"
              >
                Foo
              </option>,
              <option
                value="bar"
              >
                Bar
              </option>,
            ],
            "onChange": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "Foo",
                "value": "foo",
              },
              "ref": null,
              "rendered": "Foo",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "Bar",
                "value": "bar",
              },
              "ref": null,
              "rendered": "Bar",
              "type": "option",
            },
          ],
          "type": "select",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Select/> should have label 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Select
    label="Test label"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <label>
          Test label
        </label>,
        <select
          onChange={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test label",
        },
        "ref": null,
        "rendered": "Test label",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "onChange": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "select",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <label>
            Test label
          </label>,
          <select
            onChange={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test label",
          },
          "ref": null,
          "rendered": "Test label",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "onChange": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "select",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Select/> should render options 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Select
    options={
      Array [
        Object {
          "text": "Foo",
          "value": "foo",
        },
        Object {
          "text": "Bar",
          "value": "bar",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <label />,
        <select
          onChange={[Function]}
        >
          <option
            value="foo"
          >
            Foo
          </option>
          <option
            value="bar"
          >
            Bar
          </option>
        </select>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="foo"
            >
              Foo
            </option>,
            <option
              value="bar"
            >
              Bar
            </option>,
          ],
          "onChange": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "Foo",
              "value": "foo",
            },
            "ref": null,
            "rendered": "Foo",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "Bar",
              "value": "bar",
            },
            "ref": null,
            "rendered": "Bar",
            "type": "option",
          },
        ],
        "type": "select",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <label />,
          <select
            onChange={[Function]}
          >
            <option
              value="foo"
            >
              Foo
            </option>
            <option
              value="bar"
            >
              Bar
            </option>
          </select>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value="foo"
              >
                Foo
              </option>,
              <option
                value="bar"
              >
                Bar
              </option>,
            ],
            "onChange": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "Foo",
                "value": "foo",
              },
              "ref": null,
              "rendered": "Foo",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "Bar",
                "value": "bar",
              },
              "ref": null,
              "rendered": "Bar",
              "type": "option",
            },
          ],
          "type": "select",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Select/> should render without options 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Select />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <label />,
        <select
          onChange={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "onChange": [Function],
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "select",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <label />,
          <select
            onChange={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "onChange": [Function],
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "select",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
